app/models/concerns/profile_tailoring.rb

```diff
@@ -33,4 +33,19 @@ def update_os_minor_version(version)
 
     update!(os_minor_version: version)
   end
+
+  def tailoring_valid?
+    rule_groups_for_rules = RuleGroup.where(id: rules.includes(:rule_group)
+                                     .references(:rule_group).flat_map(&:rule_group_ancestor_ids))
+
+    conflicts = relationships_for(rules, rule_groups_for_rules, 'conflicts').distinct.map(&:right)
+    requires = relationships_for(rules, rule_groups_for_rules, 'requires').distinct.map(&:right)
+
+    valid_relationships?(rules, rule_groups_for_rules, requires, conflicts)
+  end
+
+  def valid_relationships?(rules, rule_groups, requires, conflicts)
+    rules_and_groups = rules | rule_groups
```

Comment:
What does the `|` operation do?

Reply:
it concatenates the two arrays, https://ruby-doc.org/core-2.3.1/Array.html#method-i-7C

Reply:
It does a union because we still need to combine the rules and rule_groups to check them against `conflicts` and `requires`

Reply:
And why not ˙+`? I guess that's what's being used across the codebase.

Reply:
Hmm I guess I just went with `|` because I'm also using the intersect `&` operator in that method but I can use `+` to stay consistent with the rest of the codebase because the result is the same

Reply:
No, you should stick with `|`, it's not a simple concatenation:
```
[1] | [1,2] => [1,2]
```